### PR TITLE
LPD-40177 | LPD-37597 | LPD-37596 | LPD-37595 | LPD-37593 | LPD-37592 | LPD-37591 Allow exporting and importing one Page referencing

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/LPD-40177
https://liferay.atlassian.net/browse/LPD-37597
https://liferay.atlassian.net/browse/LPD-37596
https://liferay.atlassian.net/browse/LPD-37595
https://liferay.atlassian.net/browse/LPD-37593
https://liferay.atlassian.net/browse/LPD-37592
https://liferay.atlassian.net/browse/LPD-37591

@alessiorendina

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPOptionFacetsPortletInstanceConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	public String displayStyleGroupKey();
+
+	@Meta.AD(deflt = "10", name = "max-options", required = false)
+	public int maxOptions();
+
+	@Meta.AD(deflt = "10", name = "max-terms", required = false)
+	public int maxTerms();
+
+	@Meta.AD(deflt = "1", name = "frequency-threshold", required = false)
+	public int frequencyThreshold();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPOptionFacetsPortletInstanceConfiguration.java:60]: Sort alphabetically

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPSpecificationOptionFacetsPortletInstanceConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * @author Crescenzo Rega
+ */
+@ExtendedObjectClassDefinition(
+	category = "catalog",
+	scope = ExtendedObjectClassDefinition.Scope.PORTLET_INSTANCE
+)
+@Meta.OCD(
+	id = "com.liferay.commerce.product.content.search.web.internal.configuration.CPSpecificationOptionFacetsPortletInstanceConfiguration",
+	localization = "content/Language",
+	name = "commerce-product-option-facets-portlet-instance-configuration-name"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPSpecificationOptionFacetsPortletInstanceConfiguration.java:23]: `commerce-product-option-facets` -> `commerce-product-specification-option-facets`

Comment (alessiorendina) [reply modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPSpecificationOptionFacetsPortletInstanceConfiguration.java:23]: Remember to add the language key as well

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPSpecificationOptionFacetsPortletInstanceConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	@Meta.AD(deflt = "10", name = "max-terms", required = false)
+	public int maxTerms();
+
+	@Meta.AD(deflt = "1", name = "frequency-threshold", required = false)
+	public int frequencyThreshold();
+
+	@Meta.AD(
+		deflt = "priority:asc", name = "specifications-order", required = false
+	)
+	public String specificationsOrder();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/configuration/CPSpecificationOptionFacetsPortletInstanceConfiguration.java:65]: Sort alphabetically

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPSpecificationOptionsFacetDisplayContextBuilder.java [hunk 4]
```diff
@@ -286,24 +281,37 @@ public void renderRequest(RenderRequest renderRequest) {
...
 		int minCount = 1;
 
-		if (_frequenciesVisible &&
-			_displayStyle.equals(
+		boolean frequenciesVisible =
+			_cpSpecificationOptionFacetsPortletInstanceConfiguration.
+				frequenciesVisible();
+		String displayStyle =
+			_cpSpecificationOptionFacetsPortletInstanceConfiguration.
+				displayStyle();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPSpecificationOptionsFacetDisplayContextBuilder.java:289]: Inline those 2

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPSpecificationOptionsFacetDisplayContextBuilder.java [hunk 5]
```diff
@@ -286,24 +281,37 @@ public void renderRequest(RenderRequest renderRequest) {
...
+		int maxTerms =
+			_cpSpecificationOptionFacetsPortletInstanceConfiguration.maxTerms();
+		int frequencyThreshold =
+			_cpSpecificationOptionFacetsPortletInstanceConfiguration.
+				frequencyThreshold();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPSpecificationOptionsFacetDisplayContextBuilder.java:294]: Sort alphabetically

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPOptionsSearchFacetDisplayContextBuilder.java [hunk 6]
```diff
@@ -130,24 +123,33 @@ protected CPOptionsSearchFacetTermDisplayContext buildTermDisplayContext(
...
-		if (_frequenciesVisible &&
-			_displayStyle.equals(
-				"ddmTemplate_CP-SPECIFICATION-OPTION-FACET-CLOUD-FTL")) {
+		boolean frequenciesVisible =
+			_cpOptionFacetsPortletInstanceConfiguration.frequenciesVisible();
+		String displayStyle =
+			_cpOptionFacetsPortletInstanceConfiguration.displayStyle();
+		int maxTerms = _cpOptionFacetsPortletInstanceConfiguration.maxTerms();
+		int frequencyThreshold =
+			_cpOptionFacetsPortletInstanceConfiguration.frequencyThreshold();
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/builder/CPOptionsSearchFacetDisplayContextBuilder.java:132]: Same here

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPOptionsSearchFacetDisplayContext.java [hunk 7]
```diff
@@ -31,18 +35,26 @@
...
 
+		_groupLocalService = groupLocalService;
 		_httpServletRequest = httpServletRequest;
 
 		_cpRequestHelper = new CPRequestHelper(httpServletRequest);
 
 		_renderRequest = _cpRequestHelper.getRenderRequest();
+
+		_cpOptionFacetsPortletInstanceConfiguration =
+			ConfigurationProviderUtil.getPortletInstanceConfiguration(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPOptionsSearchFacetDisplayContext.java:55]: Pass ConfigurationProvider into constructor instead

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java [hunk 8]
```diff
@@ -24,10 +30,25 @@
...
 			HttpServletRequest httpServletRequest)
 		throws ConfigurationException {
 
+		_groupLocalService = groupLocalService;
 		_httpServletRequest = httpServletRequest;
+
+		_cpRequestHelper = new CPRequestHelper(httpServletRequest);
+
+		_cpSpecificationOptionFacetsPortletInstanceConfiguration =
+			ConfigurationProviderUtil.getPortletInstanceConfiguration(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java:43]: Same here

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/BaseCPPublisherDisplayContext.java [hunk 9]
```diff
@@ -22,6 +22,9 @@
 import com.liferay.petra.string.StringPool;
 import com.liferay.portal.configuration.module.configuration.ConfigurationProviderUtil;
 import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Group;
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/BaseCPPublisherDisplayContext.java:25]: Same thing for ConfigurationProvider

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentDisplayContext.java [hunk 10]
```diff
@@ -25,6 +25,9 @@
 import com.liferay.portal.configuration.module.configuration.ConfigurationProviderUtil;
 import com.liferay.portal.kernel.cookies.CookiesManagerUtil;
 import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Group;
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentDisplayContext.java:28]: Same thing for ConfigurationProvider

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentMiniDisplayContext.java [hunk 11]
```diff
@@ -26,6 +26,9 @@
 import com.liferay.portal.configuration.module.configuration.ConfigurationProviderUtil;
 import com.liferay.portal.kernel.cookies.CookiesManagerUtil;
 import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Group;
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPCompareContentMiniDisplayContext.java:29]: Same thing for ConfigurationProvider

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPContentConfigurationDisplayContext.java [hunk 12]
```diff
@@ -6,9 +6,13 @@
 package com.liferay.commerce.product.content.web.internal.display.context;
 
 import com.liferay.commerce.product.content.web.internal.configuration.CPContentPortletInstanceConfiguration;
+import com.liferay.petra.string.StringPool;
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/display/context/CPContentConfigurationDisplayContext.java:9]: Same thing for ConfigurationProvider

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSpecificationOptionFacetPortletConfigurationAction.java [hunk 13]
```diff
@@ -33,16 +35,16 @@
 	property = "javax.portlet.name=" + CPPortletKeys.CP_SPECIFICATION_OPTION_FACETS,
 	service = ConfigurationAction.class
 )
-public class CPSpecificationtOptionFacetPortletConfigurationAction
-	extends DefaultConfigurationAction {
+public class CPSpecificationOptionFacetPortletConfigurationAction
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSpecificationOptionFacetPortletConfigurationAction.java:38]: `Facet` -> `Facets`

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSearchResultsConfigurationAction.java [hunk 14]
```diff
@@ -33,7 +33,7 @@
 	service = ConfigurationAction.class
 )
 public class CPSearchResultsConfigurationAction
-	extends DefaultConfigurationAction {
+	extends BaseConfigurationAction {
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/action/CPSearchResultsConfigurationAction.java:36]: Move all modified in an "update usage" commit

File: modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPOptionFacetsPortlet.java [hunk 15]
```diff
@@ -10,6 +10,7 @@
 import com.liferay.commerce.product.content.search.web.internal.display.context.builder.CPOptionsSearchFacetDisplayContextBuilder;
 import com.liferay.commerce.product.service.CPOptionLocalService;
 import com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet;
+import com.liferay.portal.kernel.service.GroupLocalService;
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPOptionFacetsPortlet.java:13]: Add this commit to the "update usage" commit

File: modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/option_facets/configuration.jsp [hunk 16]
```diff
@@ -8,7 +8,7 @@
 <%@ include file="/init.jsp" %>
 
 <%
-CPOptionsSearchFacetDisplayContext cpOptionsSearchFacetDisplayContext = new CPOptionsSearchFacetDisplayContext(request);
+CPOptionsSearchFacetDisplayContext cpOptionsSearchFacetDisplayContext = (CPOptionsSearchFacetDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-search-web/src/main/resources/META-INF/resources/option_facets/configuration.jsp:11]: Move this commit to the "update usage" commit as well

File: modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/compare_products/configuration/adt.jsp [hunk 17]
```diff
@@ -16,7 +16,7 @@ CPCompareContentDisplayContext cpCompareContentDisplayContext = (CPCompareConten
 		<liferay-template:template-selector
 			className="<%= CPCompareContentPortlet.class.getName() %>"
 			displayStyle="<%= cpCompareContentDisplayContext.getDisplayStyle() %>"
-			displayStyleGroupId="<%= cpCompareContentDisplayContext.getDisplayStyleGroupId() %>"
+			displayStyleGroupKey="<%= cpCompareContentDisplayContext.getDisplayStyleGroupKey() %>"
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/resources/META-INF/resources/compare_products/configuration/adt.jsp:19]: Move this commit to the "update usage" commit

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@CrescenzoRegaLF LGTM overall, just few small changes. Thank you!

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:report:87319667

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:report:87837073

**Comment by CrescenzoRegaLF** [comment]
ci:test:upgrade

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:upgrade

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Pull request closed without merging**